src/wstool/multiproject_cmd.py

```diff
@@ -274,6 +276,32 @@ def do_work(self):
     return outputs
 
 
+def cmd_foreach(config, command, shell=False):
+    """Run command in all SCM entries in config, relative to path"""
+
+    class ForeachRetriever(object):
+        def __init__(self, element, command, shell):
+            self.element = element
+            self.command = command
+            self.shell = shell
+
+        def do_work(self):
+            command = self.command
+            if not self.shell:
+                command = shlex.split(command)
+            po = Popen(command, cwd=self.element.path, shell=self.shell,
+                       stdout=PIPE, stderr=PIPE)
+            stdout, stderr = po.communicate()
+            return {'stdout': stdout, 'stderr': stderr}
+
+    elements = config.get_config_elements()
+    work = DistributedWork(capacity=len(elements), num_threads=-1)
```

Comment:
not sure about the constant -1 here,  maybe a conservative value would be better. We use -1 if we know the workers will not do CPU intensive things.